Reference/Routing/Request-Pipeline/IContentFinder.md

```diff
@@ -180,33 +179,37 @@ namespace RoutingDocs.ContentFinders
             // Find the root node with a matching domain to the incoming request
             var allDomains = _domainService.GetAll(true).ToList();
             var domain = allDomains?
-                .FirstOrDefault(f => f.DomainName == contentRequest.Uri.Authority 
-                || f.DomainName == $"https://{contentRequest.Uri.Authority}"
-                || f.DomainName == $"http://{contentRequest.Uri.Authority}");
-                
+                .FirstOrDefault(f => f.DomainName == contentRequest.Uri.Authority
+                                     || f.DomainName == $"https://{contentRequest.Uri.Authority}"
+                                     || f.DomainName == $"http://{contentRequest.Uri.Authority}");
+
             var siteId = domain != null ? domain.RootContentId : allDomains.Any() ? allDomains.FirstOrDefault()?.RootContentId : null;
 
-            if(!_umbracoContextAccessor.TryGetUmbracoContext(out var umbracoContext))
+            if (!_umbracoContextAccessor.TryGetUmbracoContext(out var umbracoContext))
             {
-                return false;
+                return Task.FromResult(false);
             }
+
+            if (umbracoContext.Content == null)
+                return new Task<bool>(() => contentRequest.PublishedContent is not null);
+
             var siteRoot = umbracoContext.Content.GetById(false, siteId ?? -1);
 
             if (siteRoot is null)
             {
-                return false;
+                return Task.FromResult(false);
             }
 
             // Assuming the 404 page is in the root of the language site with alias fourOhFourPageAlias
-            var notFoundNode = siteRoot.Children.FirstOrDefault(f => f.ContentType.Alias == "fourOhFourPageAlias");
+            var notFoundNode = siteRoot.Children?.FirstOrDefault(f => f.ContentType.Alias == "fourOhFourPageAlias");
 
-            if (notFoundNode is not null)
+            if (notFoundNode != null)
```

Comment:
Not sure about this.. `!=` is less effective then `is not` due to possible operation overloads..